### PR TITLE
Fixed several issues found by OSS-Fuzz

### PR DESCRIPTION
**Pull request by nikitamikhaylov**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixed several issues found by OSS-Fuzz.

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
This is an automated comment for commit af603c2cc6455b31aba1a70c967c35b083fe6c0a with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/51736/af603c2cc6455b31aba1a70c967c35b083fe6c0a/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: src/Core/Field.h [hunk 1]
```diff
@@ -671,6 +677,27 @@ class Field
 
     Types::Which which;
 
+    /// Field may contain a Field inside in case when Field stores Array, Tuple, Map or Object.
+    /// As the result stack overflow on destruction is possible
+    /// and to avoid it we need to count the depth and have a threshold.
+    size_t nested_field_depth = 0;
```
Comment (alexey-milovidov) [inline src/Core/Field.h:683]: But it increases the size of all Fields.

Comment (nikitamikhaylov) [reply src/Core/Field.h]: Now it will increase the size of only Array, Map, Object and Tuple. They are already big enough.

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
Could you please also post a reproducer here?
I will think if there is a way to solve it more simply...

**Comment by nikitamikhaylov** [comment]
[repro.txt](https://github.com/ClickHouse/ClickHouse/files/11948544/repro.txt)

Original stacktrace looks like this https://pastila.nl/?02a26922/17fbdcc175fb60cef22a004887ccd170

**Pull request merged**